src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/sensors/Sensors.scala

```diff
@@ -43,6 +43,7 @@ class Sensors @Inject()(eventProcessor: EventProcessor, sensorRepository: Sensor
   def processEvents(): Future[Unit] = {
     logger.debug(s"Processing events. Sensors: ${sensors.keys}")
     val fut = for {
+      _ <- updateChangedSensors()
```

Comment:
I would change order. _ <- removeInactiveSensors() and after that  _ <- updateChangedSensors()